src/plugins/intel_gna/tests/unit/pre_post_processing/gna_output_postproc_test.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <map>
+#include <vector>
+
+#include "any_copy.hpp"
+#include "common_test_utils/data_utils.hpp"
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "gna_data_types.hpp"
+#include "gna_plugin.hpp"
+#include "pre_post_process/converter_factory.hpp"
+#include "pre_post_process/input_output_data_handler.hpp"
+#include "pre_post_process/preprocessing.hpp"
+
+using namespace InferenceEngine;
+
+namespace testing {
+
+typedef std::tuple<InferenceEngine::Precision,   // input precision
+                   InferenceEngine::Precision,   // output precision
+                   InferenceEngine::SizeVector,  // input shape
+                   intel_dnn_orientation_t,      // orientation
+                   float,                        // scale factor
+                   bool,                         // test avx2
+                   uint32_t                      // input range
+                   >
+    GNAOutputPrecisionParams;
+
+template <typename U, typename T>
+class GNAOutputPrecisionTest : public ::testing::TestWithParam<GNAOutputPrecisionParams> {
+public:
+    void SetUp() override {
+        uint32_t input_range;
+        std::tie(precisionIn, precisionOut, shape, orientation, sf, testAvx2, input_range) = GetParam();
+        inputValues.resize(ov::shape_size(shape));
+        CommonTestUtils::fill_data_random(&inputValues[0], ov::shape_size(shape), input_range);
+
+        std::transform(begin(inputValues), end(inputValues), std::back_inserter(referenceValues), [this](U i) {
+            return static_cast<T>(i / sf);
+        });
+
+        if (testAvx2) {
+            pre_post_processing::ConverterFactory converter_factory;
+            m_input_output_handler.set_data_converter(converter_factory.create_converter());
```

Comment:
please remove set_data_converter from hanlder
you can control this via constructor if m_input_output_handler if you will use this as shared_ptr